src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -174,87 +226,162 @@ struct boss_jeklik : public BossAI
                     break;
                 case EVENT_SPAWN_BATS:
+                    LOG_DEBUG("scripts.ai", "Jeklik: EVENT_SPAWN_BATS");
                     Talk(EMOTE_SUMMON_BATS);
                     if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
                         for (uint8 i = 0; i < 6; ++i)
                             if (Creature* bat = me->SummonCreature(NPC_BLOODSEEKER_BAT, SpawnBat[i], TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
                                 bat->AI()->AttackStart(target);
                     events.ScheduleEvent(EVENT_SPAWN_BATS, 30s, PHASE_ONE);
                     break;
-                    //Phase two
+                // PHASE_TWO
                 case EVENT_CURSE_OF_BLOOD:
+                    LOG_DEBUG("scripts.ai", "Jeklik: EVENT_CURSE_OF_BLOOD");
                     DoCastSelf(SPELL_CURSE_OF_BLOOD);
                     events.ScheduleEvent(EVENT_CURSE_OF_BLOOD, 25s, 30s, PHASE_TWO);
                     break;
                 case EVENT_PSYCHIC_SCREAM:
+                    LOG_DEBUG("scripts.ai", "Jeklik: EVENT_PSYCHIC_SCREAM");
                     DoCastVictim(SPELL_PSYCHIC_SCREAM);
                     events.ScheduleEvent(EVENT_PSYCHIC_SCREAM, 35s, 45s, PHASE_TWO);
                     break;
                 case EVENT_SHADOW_WORD_PAIN:
+                    LOG_DEBUG("scripts.ai", "Jeklik: EVENT_SHADOW_WORD_PAIN");
                     DoCastRandomTarget(SPELL_SHADOW_WORD_PAIN, 0, true);
                     events.ScheduleEvent(EVENT_SHADOW_WORD_PAIN, 12s, 18s, PHASE_TWO);
                     break;
                 case EVENT_MIND_FLAY:
+                    LOG_DEBUG("scripts.ai", "Jeklik: EVENT_MIND_FLAY");
                     DoCastVictim(SPELL_MIND_FLAY);
                     events.ScheduleEvent(EVENT_MIND_FLAY, 20s, 40s, PHASE_TWO);
                     break;
                 case EVENT_GREATER_HEAL:
+                    LOG_DEBUG("scripts.ai", "Jeklik: EVENT_GREATER_HEAL");
                     Talk(EMOTE_GREAT_HEAL);
                     me->InterruptNonMeleeSpells(false);
                     DoCastSelf(SPELL_GREATER_HEAL);
                     events.ScheduleEvent(EVENT_GREATER_HEAL, 25s, PHASE_TWO);
                     break;
                 case EVENT_SPAWN_FLYING_BATS:
-                    Talk(SAY_CALL_RIDERS);
-                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
-                        if (Creature* flyingBat = me->SummonCreature(NPC_FRENZIED_BAT, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ() + 15.0f, 0.0f, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
-                            flyingBat->AI()->DoZoneInCombat();
-                    events.ScheduleEvent(EVENT_SPAWN_FLYING_BATS, 10s, 15s, PHASE_TWO);
+                    LOG_DEBUG("scripts.ai", "Jeklik: EVENT_SPAWN_FLYING_BATS");
+                    if (SelectTarget(SelectTargetMethod::Random, 0))
```

Comment:
Correct me if I'm wrong, but this feels like it's just checking if Jeklik is in combat, in which case I feel like there must be a better way to accomplish it than this.

Reply:
Close - engaging the boss puts the entire zone in combat. However, I believe `SelectTarget` only pulls from the list of players on the threat table.

I did just follow the lead of the original author here, but I'm glad to pivot if checking just for InCombat is better.

Reply:
Would this be clearer? I'll need to test it, of course.

`if (me->GetThreatMgr().GetThreatListSize() > 0)`

Reply:
I'm just asking from a performant standpoint. The previous implementation actually used the selected target as a part of the script, whereas this one just seems, like I said, to check for valid combat targets.

> Would this be clearer?

Might be able to knock off the `> 0` and the evaluation will work the same, but yes, it does seem clearer to me at least.